### ISSUE TITLE
VM list does not match on PVE 8 version

### ISSUE THREAD

**Issue by rozaqi** [Starttoaster/proxmox-exporter#50]
Hi all,

i found vm list mismatch issue in proxmox version 8.2.2. I tried to reproduce the issue on version 7.3-6 but it didn't happen.

After I tried debugging it turned out that there was a VM that was entered into the template and it shouldn't have been entered.

## Debug
![image](https://github.com/user-attachments/assets/6e768ec7-e9ec-4c43-872a-dd729a109539)

## log
![image](https://github.com/user-attachments/assets/310a9da1-a29f-43ae-95ce-9a6952ef503f)

## VM
![image](https://github.com/user-attachments/assets/d9d1aad2-4465-4ba1-a4a7-26d9150bf15b)

## VM Config
![image](https://github.com/user-attachments/assets/5f6b9450-fefc-4145-8191-3a714c0e1548)

Thank You.

**Comment by rozaqi** [comment]
Once I tried analyzing the source code to check if the VM was a template, it didn't seem quite right. because the VM name is the same as the template name it is detected as a template.

![image](https://github.com/user-attachments/assets/4832dc3e-636a-434d-b5c8-64308623aa19)

**Comment by Starttoaster** [comment]
Hi, thanks for raising an issue!

> because the VM name is the same as the template name it is detected as a template.

Just to confirm, you're saying here that you have a VM template with the same name as an actual VM in your cluster?

**Comment by rozaqi** [comment]
yes I have the same vm name as the template name, I've tried editing the source code but still can't fix it, I hope this helps, usually I use the following command to get a list of VM templates:
```
pvesh get /cluster/resources --output-format json | jq '[.[] | select(.type == "qemu" and .template == 1) | {name: .name, vmid: .vmid}]'
```

**Comment by Starttoaster** [comment]
Thanks for the confirmation! I suppose when I was writing that, I had the assumption that most people name their VMs with a purpose rather than a distro. For example my kubernetes VMs are all named controller1-3 and worker1-3, but I think it's fair to cover this case. I'll see what I can do to get a patch for that hopefully later today. The first thing that comes to mind would just be to compare VM ID to VM ID, rather than VM names.

**Comment by Starttoaster** [comment]
Hey there! This is solved in v0.2.2. Thanks again for the report! :)
